pkg/kubelet/dockershim/libdocker/fake_client.go

```diff
@@ -145,6 +149,24 @@ func (f *FakeDockerClient) appendContainerTrace(traceCategory string, containerN
 	}
 }
 
+// WriteDockerConfigFile creates a config file that NewDockerKeyring will pick up. It returns
+// a function that will delete the file, so the caller can `defer` it.
+func (f *FakeDockerClient) WriteDockerConfigJSONFile(cfg string) (error, func()) {
```

Comment:
I'm with you about the reservations.

Do you mean refactor `ensureSandboxImageExists` into a thin shim, with a new `ensureSandboxImageExistsWithPaths` function containing all the code ? Works for me.

Alternatively: we could use a package variable to hold the searchpath (`defaultDockercfgSearchPath`), and then overwrite it with the tmpdir during the test.